### PR TITLE
[Reviewer: Ellie] Add ENT logs to crest

### PR DESCRIPTION
**Pull request by mirw**
Ellie,

Please can you review my addition of ENT logs to crest?  This also depends on the python-common changes in https://github.com/Metaswitch/python-common/pull/28.

I've run `make verify`, `make test` and live tested.

Thanks,

Matt

### CODE REVIEW

**Comment by mirw** [comment]
I've removed some of the request-scope logs, and tweaked the wording to match Sprout logs.

File: src/metaswitch/crest/api/base.py [hunk 1]
```diff
@@ -215,7 +223,7 @@ def setupStats(p_id, worker_proc):
 def shutdownStats():
     zmq.unbind()
 
-def _guess_mime_type(body):
+def _guess_mime_type(body, client_ip):
```
Comment (eleanor-merry) [inline src/metaswitch/crest/api/base.py]: Is client_ip used anywhere?

File: src/metaswitch/crest/main.py [hunk 2]
```diff
@@ -72,10 +74,25 @@ def bind_safely(reactor, process_id, application):
...
+    text = twisted.python.log.textFromEventDict(eventDict)
+    if text is None:
+        return
+    if eventDict['isError']:
+        level = logging.ERROR
+    elif 'level' in eventDict:
+        level = eventDict['level']
+    else:
+        level = settings.LOG_LEVEL
+    if level >= logging.ERROR:
```
Comment (eleanor-merry) [inline src/metaswitch/crest/main.py]: Why does the PD log depend on the logging level?

File: src/metaswitch/crest/pdlogs.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    desc="Service '%s' is starting." % settings.LOG_FILE_PREFIX,
+    cause="A 'start' was requested by an external entity.",
+    effect="%s service is starting." % settings.LOG_FILE_PREFIX,
+    action="None.",
+    priority=PDLog.LOG_NOTICE)
+
+CREST_UP = PDLog(
+    number=PDLog.CL_CREST_ID + 3,
+    desc="Service '%s' is up and listening for UNIX socket." % (settings.LOG_FILE_PREFIX),
+    cause="A shutdown was requested by an external entity.",
```
Comment (eleanor-merry) [inline src/metaswitch/crest/pdlogs.py]: Cause is incorrect

**Comment by eleanor-merry** [comment]
Looks good, a few comments inline

**Pull request merged**